src/main/scala/loamstream/loam/intake/RowDef.scala

```diff
@@ -7,3 +7,7 @@ package loamstream.loam.intake
 final case class RowDef(varIdDef: SourcedColumnDef, otherColumns: Seq[SourcedColumnDef]) {
   def columnDefs: Seq[SourcedColumnDef] = varIdDef +: otherColumns
 }
+
+final case class UnsourcedRowDef(varIdDef: UnsourcedColumnDef, otherColumns: Seq[UnsourcedColumnDef]) {
```

Comment:
Just noting that comments would be quite helpful. I don't know the definition of a `SourcedColumnDef` so an `UnsourcedColumnDef` is equally confusing. Does source mean a filename, a column name, a column index, something else? 

I don't know exactly where this code would be used. But, if I was Marcin writing an intake loam config, and I had to pick between these classes, how would I know which to use?

Reply:
If you were Marcin, you wouldn't reference one of these directly.

I didn't document the DSL much because it was a sketch that became a prototype, and for a while it wasn't at all clear that we'd keep it.  Now it looks like we are, though.

As for these classes, the `ColumDef`s are basically functions that take rows from CSV files and extract/compute values from them.  `RowDef`s are collections of `ColumnDef`s that transform a CSV row into another CSV row by extracting/computing multiple values.  

Sourced/Unsourced means there is an associated `CsvSource` or not.  A `CsvSource` is an abstraction for producing `CsvRow`s from something: a file, a Bash pipeline, etc.